bentoml/yatai/client/bento_repository_api.py

```diff
@@ -146,6 +146,38 @@ def _upload_bento_service(self, bento_service, saved_bento_path):
 
             return response.uri.uri
 
+        elif response.uri.type == BentoUri.GCS:
+            self._update_bento_upload_progress(
+                bento_service_metadata, UploadStatus.UPLOADING, 0
+            )
+
+            fileobj = io.BytesIO()
+            with tarfile.open(mode="w:gz", fileobj=fileobj) as tar:
+                tar.add(saved_bento_path, arcname=bento_service_metadata.name)
+            fileobj.seek(0, 0)
+
+            http_response = requests.put(response.uri.cloud_presigned_url, data=fileobj)
+
+            if http_response.status_code != 200:
+                self._update_bento_upload_progress(
+                    bento_service_metadata, UploadStatus.ERROR
+                )
+                raise BentoMLException(
+                    f"Error saving BentoService bundle to GCS. "
+                    f"{http_response.status_code}: {http_response.text}"
+                )
+
+            self._update_bento_upload_progress(bento_service_metadata)
+
+            logger.info(
+                "Successfully saved BentoService bundle '%s:%s' to GCS: %s",
+                bento_service_metadata.name,
+                bento_service_metadata.version,
+                response.uri.uri,
+            )
+
+            return response.uri.uri
+
```

Comment:
This seems like using the same code path as the S3 one.  Let's refactor this.

Reply:
@yubozhao Would you like us to merge this `if` condition with S3 `if` condition?

Reply:
Yes.  The only difference between those two statements is raising exceptions. We should be refactoring them